src/basilisp/lang/obj.py

```diff
@@ -76,17 +77,46 @@ def map_lrepr(
     with the start and end string supplied. The entries argument must be a
     callable which will produce tuples of key-value pairs.
 
+    If the keyword argument print_namespace_maps is True and all keys
+    are symbols or keywords and they share the same namespace, then
+    print the namespace of the keys at the beginning of the map
+    instead of beside the keys.
+
     The keyword arguments will be passed along to lrepr for the sequence
-    elements."""
+    elements.
+
+    """
+    from basilisp.lang import keyword as kw  # pylint: disable=cyclic-import
+    from basilisp.lang import symbol as sym  # pylint: disable=cyclic-import
```

Comment:
I believe the `INamed` interface I created in #884 should suffice to resolve the need for these circular imports.

Reply:
Hi, and thanks for looking into this.

could you please elaborate a bit how to use the new `INamed` interface.  `basilisp.lang.interfaces` imports `obj.py` and as such if I try to use the same logic there's still a circular dependency (down from two) by  importing `interfaces.py` inline from `object.py`.

Furthermore, with the current logic I need to strip the ns from the symbol or keyword instance, by creating a new instance without it. If I only have the `INamed` interface at hand, i don't think I am able to create an instance of the same type ? for example

currently
``` python
    def entry_reprs():
        for k, v in entries():
            key = k
            if key_ns_drop:
                if isinstance(k, kw.Keyword):
                    key = kw.Keyword(k.name)
                elif isinstance(k, sym.Symbol):
                    key = sym.Symbol(k.name)
            yield f"{lrepr(key, **kwargs)} {lrepr(v, **kwargs)}"
```

with INamed interface
``` python
    def entry_reprs():
        for k, v in entries():
            key = k
            if key_ns_drop:
                if isinstance(k, INamed):
                    key = ?
            yield f"{lrepr(key, **kwargs)} {lrepr(v, **kwargs)}"
```

thanks

Reply:
I think the change in #886 should support this most recent issue:

```python
    def entry_reprs():
        for k, v in entries():
            key = k
            if key_ns_drop:
                if isinstance(k, INamed):
                    key = k.with_name(k.name)
            yield f"{lrepr(key, **kwargs)} {lrepr(v, **kwargs)}"
```

As to the circular dependency, you should be able to move `map_lrepr` over to `src/basilisp/lang/map.py` since that is the only place it is used.